cmd/cmdtrace/cmd_span.go

```diff
@@ -44,13 +46,17 @@ import (
 // command invocation to ensure the span is properly finalized and
 // exported before exit.
 func Setup(cmd *cobra.Command, dockerCli command.Cli, args []string) error {
+	if v, _ := strconv.ParseBool(os.Getenv("COMPOSE_EXPERIMENTAL_OTEL")); !v {
+		return nil
+	}
+
 	tracingShutdown, err := tracing.InitTracing(dockerCli)
 	if err != nil {
 		return fmt.Errorf("initializing tracing: %w", err)
 	}
 
 	ctx := cmd.Context()
-	ctx, cmdSpan := tracing.Tracer.Start(
+	ctx, cmdSpan := tracing.TracingProvider.Tracer("compose").Start(
```

Comment:
must not set Tracer name to "compose"

> 	// The passed name needs to uniquely identify instrumentation code.
>	// Therefore, it is recommended that name is the Go package name of the
>	// library providing instrumentation (note: not the code being
>	// instrumented)

yes, this is crazy and terribly confusing, instrumented code is identified by `service.name` attribute, ... which we set in compose to track compose service name 😓, need to be fixed

Reply:
😭